packages/teraslice/test/lib/cluster/services/cluster/backends/kubernetes/net-spec.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const net = require('net');
+const Porty = require('porty');
+
+const {
+    waitForTcpPortOpen
+} = require('../../../../../../../lib/cluster/services/cluster/backends/kubernetes/net');
+
+// Run tests with DEBUG=True to see output
+function l(msg) {
+    if (process.env.DEBUG) {
+        // eslint-disable-next-line no-console
+        console.log(msg);
+    }
+}
+
+const logger = {
+    debug: (msg) => l(`[DEBUG]: ${msg}`),
+    error: (msg) => l(`[ERROR]: ${msg}`),
+    info: (msg) => l(`[INFO]: ${msg}`),
+};
+
+let server;
+let serverPort;
+
+async function setServerPort() {
+    serverPort = await Porty.find({ min: 14500, max: 14549 });
+    logger.debug(`serverPort: ${serverPort}`);
+}
+
+async function startServer() {
+    server = await net.createServer();
+    await server.listen(serverPort, () => { });
+}
+
+async function stopServer() {
+    await server.close();
+}
+
+describe('waitForTcpPortOpen', () => {
+    describe('with service running', () => {
+        beforeEach(async () => {
+            await setServerPort();
+            await startServer();
+        });
+        afterEach(async () => { await stopServer(); });
+
+        test('should not raise an error', async () => {
+            const options = {
+                host: 'localhost',
+                logger,
+                port: serverPort,
+                retryFrequency: 100,
+                retryTimeout: 2000,
+            };
+            await expect(waitForTcpPortOpen(options)).resolves.toBe();
```

Comment:
I prefer this matcher since it is more clear.

```suggestion
            await expect(waitForTcpPortOpen(options)).toResolve();
```

Reply:
Oh, yeah, thanks!  That was all new stuff to me, I was just pleased to figure out how to do it at all.